src/main/java/seedu/canoe/logic/commands/AddStudentToTrainingCommand.java

```diff
@@ -171,11 +167,27 @@ public boolean uniqueChecker(Training trainingToCheck, Student check) throws Com
         return true;
     }
 
+    /**
+     * Checks if the whether the specified Student is available at the Date
+     * and Time of the specified Training
+     * and whether the Student already has an Attendance at that Date and Time
+     * @param training Training to be attended
+     * @param student Student to check
+     * @return boolean that indicates whether Student to be added is available.
+     */
+    public boolean isStudentAvailable(Training training, Student student) {
+        if (!student.isAvailableAtDateTime(training.getDateTime())
+                || student.hasAttendanceAtDateTime(training.getDateTime())) {
+            return false;
+        }
+        return true;
+    }
+
     /**
      * Returns the Student object in the model with the Id same as the specified Unique Id.
      * @param model
      * @param id
-     * @return
+     * @return the Student Object that corresponds to the specified Id.
      */
     public Student getStudentWithID(Model model, String id) throws CommandException {
         id.trim();
```

Comment:
Should be `id = id.trim()`
`trim()` returns a new String, does not modify the old String.